src/Models/DataType.php

```diff
@@ -141,7 +142,17 @@ public function fieldOptions()
     {
         $table = $this->name;
 
-        $fieldOptions = SchemaManager::describeTable($table);
+        // Get ordered BREAD fields
+        $orderedFields = $this->rows()->pluck('field')->toArray();
+
+        $_fieldOptions = SchemaManager::describeTable($table)->toArray();
+
+        $fieldOptions = [];
+        $f_size = count($orderedFields);
+        for ($i = 0; $i < $f_size; $i++) {
```

Comment:
I think we are loosing new fields here (added through DB edit but not yet in data_rows)